File: python/树/1469. 寻找所有的独生节点.py
from typing import List
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def getLonelyNodes(self, root: TreeNode) -> List[int]:
        res = []
        def dfs(root):
            if not root:
                return
            if not root.left and root.right:
                res.append(root.right.val)
            elif not root.right and root.left:
                res.append(root.left.val)
            dfs(root.left)
            dfs(root.right)
        return res

File: go/tree/1469. 寻找所有的独生节点.go
package tree

func getLonelyNodes(root *TreeNode) []int {
	var dfs func(root *TreeNode)
	var res []int
	dfs = func(root *TreeNode) {
		if root == nil {
			return
		}
		if root.Left != nil && root.Right == nil {
			res = append(res, root.Left.Val)
		} else if root.Right != nil && root.Left == nil {
			res = append(res, root.Right.Val)
		}
		dfs(root.Left)
		dfs(root.Right)
	}
	dfs(root)
	return res
}
